### PR TITLE
Updated npm modules and added yarn support

### PR DESCRIPTION
**Pull request by kumarabhishek**
### Changes
- Updated dev dependencies:
	- eslint: ^4.1.1
	- style-loader: ^0.18.2
	- Webpack: ^3.1.0
- Adding yarn support with yarn.lock
- Configured eslint for JSX, React and Jasmine.
- Code changes are done to pass eslint error/warnings

### Eslint output before code change

```sh
yarn lint v0.27.5
$ eslint src --fix

/Users/kumarabhishek/dev/react-inspector/src/dom-inspector/DOMNodePreview.js
   1:17  error  'Component' is defined but never used  no-unused-vars
  58:39  error  'name' is defined but never used       no-unused-vars

/Users/kumarabhishek/dev/react-inspector/src/index.js
  18:17  error  'Component' is defined but never used  no-unused-vars

/Users/kumarabhishek/dev/react-inspector/src/object-inspector/ObjectInspector.js
  1:28  error  'Children' is defined but never used  no-unused-vars

/Users/kumarabhishek/dev/react-inspector/src/object-inspector/ObjectInspector.spec.js
  2:8  error  'PropTypes' is defined but never used              no-unused-vars
  9:7  error  'defaultProps' is assigned a value but never used  no-unused-vars

/Users/kumarabhishek/dev/react-inspector/src/object-inspector/ObjectLabel.js
  1:17  error  'Component' is defined but never used  no-unused-vars

/Users/kumarabhishek/dev/react-inspector/src/object-inspector/ObjectPreview.js
  1:17  error  'Component' is defined but never used  no-unused-vars

/Users/kumarabhishek/dev/react-inspector/src/object-inspector/ObjectRootLabel.js
  1:17  error  'Component' is defined but never used  no-unused-vars
  2:8   error  'PropTypes' is defined but never used  no-unused-vars

/Users/kumarabhishek/dev/react-inspector/src/object/ObjectName.js
  1:17  error  'Component' is defined but never used  no-unused-vars

/Users/kumarabhishek/dev/react-inspector/src/object/ObjectName.spec.js
   2:8  error  'PropTypes' is defined but never used              no-unused-vars
  13:7  error  'defaultProps' is assigned a value but never used  no-unused-vars

/Users/kumarabhishek/dev/react-inspector/src/object/ObjectValue.js
  1:17  error  'Component' is defined but never used  no-unused-vars

/Users/kumarabhishek/dev/react-inspector/src/object/ObjectValue.spec.js
    2:8  error    'PropTypes' is defined but never used     no-unused-vars
  119:3  warning  Duplicate spec: "should render a symbol"  jasmine/no-spec-dupes

/Users/kumarabhishek/dev/react-inspector/src/styles/ThemeProvider.js
  1:8  error  'React' is defined but never used  no-unused-vars

/Users/kumarabhishek/dev/react-inspector/src/table-inspector/DataContainer.js
  1:17  error  'Component' is defined but never used  no-unused-vars

/Users/kumarabhishek/dev/react-inspector/src/table-inspector/getHeaders.spec.js
  46:3  warning  Duplicate spec: "should work for objects whose keys are index numbers"  jasmine/no-spec-dupes

/Users/kumarabhishek/dev/react-inspector/src/table-inspector/HeaderContainer.js
  1:17  error  'Component' is defined but never used  no-unused-vars

/Users/kumarabhishek/dev/react-inspector/src/table-inspector/TableInspector.js
  68:11  error  'sorted' is assigned a value but never used  no-unused-vars

/Users/kumarabhishek/dev/react-inspector/src/tree-view/pathUtils.js
  12:62  error  'j' is defined but never used  no-unused-vars

/Users/kumarabhishek/dev/react-inspector/src/tree-view/TreeNode.js
  12:7   error  'name' is assigned a value but never used  no-unused-vars
  13:7   error  'data' is assigned a value but never used  no-unused-vars
  64:26  error  'data' is defined but never used           no-unused-vars
  64:32  error  'expanded' is defined but never used       no-unused-vars

/Users/kumarabhishek/dev/react-inspector/src/tree-view/TreeView.js
  10:5  error  Unexpected lexical declaration in case block  no-case-declarations
  10:5  error  Unexpected lexical declaration in case block  no-case-declarations
  10:5  error  Unexpected lexical declaration in case block  no-case-declarations

✖ 29 problems (27 errors, 2 warnings)

error Command failed with exit code 1.
```

### Eslint output after code change

```js
yarn lint v0.27.5
$ eslint src --fix
Done in 1.75s.
```

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
```
Comment (ndelangen) [inline .eslintrc]: Is the .js file no longer supported? I don't think so.. Why the move to a json file?

File: error.log.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
+```js
```
Comment (ndelangen) [inline error.log.md]: this is not js syntax
I think you mean `sh` (shell) or just plain text as log.

Comment (ndelangen) [reply error.log.md]: What is this log for?

**Comment by ndelangen** [comment]
Have a few minor questions @kumarabhishek, Thank you!
Overall good job! I do need to test it locally still.

**Comment by kumarabhishek** [comment]
### Changes
- Moved .eslintrc to .eslintrc.js
- Removed error.log.md

*Updated PR description use use `sh` instead of `js` in md formatting.*

**Review by ndelangen — APPROVED** [review]

**Pull request merged**